module/mempool/herocache/backdata/heropool/pool.go

```diff
@@ -29,6 +30,16 @@ type poolEntity struct {
 	node link
 }
 
+type PoolAddResult struct {
```

Comment:
usually golang encourages to use multiple return values to than grouping into a struct, in order to avoid allocating on heap. 

Grouping data into a struct is ok for other module, but I though herocache is trying to avoiding allocating memory on heap, right?

Reply:
https://github.com/onflow/flow-go/pull/2951/commits/4892426aeace2fbeb876b5f51706e8ed8089d6bd